testsuite/config_reader.py

```diff
@@ -1,13 +1,16 @@
 # pylint: disable=missing-docstring
 import configparser
-from flask import request
+from flask import request, session
 
 
-def get_config():
+def get_config(default='smart'):
     config = configparser.ConfigParser()
 
     try:
-        config.read_string(request.form['config'])
+        vendor = request.form.get('vendor', default).lower()
```

Comment:
My technique was just to pass details in the `configuration.defaults` dict
(instantiated in app.py), and read from that in config_reader. That seemed
at least somewhat less implicit/surprising (though still somewhat
implicit/surprising).

On Wed, May 18, 2016 at 9:47 AM, Erik Wiffin <notifications@github.com>
wrote:

> In testsuite/config_reader.py
> <https://github.com/sync-for-science/test-suite/pull/3#discussion_r63704862>
> :
>
> >      config = configparser.ConfigParser()
> >
> >      try:
> > -        config.read_string(request.form['config'])
> > +        vendor = request.form.get('vendor', default).lower()
>
> Yeah, it's really unfortunate, but the way behave actually invokes the
> test suite makes it really difficult to be explicit about dependencies like
> that. If I remember right, they're basically calling eval on
> environment.py. It makes sense on one hand, since it makes it impossible to
> accidentally share state between tests, but it would be really useful if I
> could pass environments into the test suite :/
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/sync-for-science/test-suite/pull/3/files/cc710a6eea17ff5139acbbcd08be15dd88d2ed6d..8cf1c676ffba730ce3352a51f2bcd6f974599adb#r63704862>
>